src/chocolatey/infrastructure.app/services/ChocolateyPackageService.cs

```diff
@@ -420,6 +420,19 @@ public void handle_package_result(PackageResult packageResult, ChocolateyConfigu
 
             remove_pending(packageResult, config);
 
+            // Test to see if exit code is either:
+            // 1641 - restart initiated
+            // 3010 - restart required
+            var rebootExitCodes = new List<int> { 1641, 3010 };
+
+            if(rebootExitCodes.Contains(packageResult.ExitCode))
+            {
+                if(config.Features.StopOnFirstPackageRebootRequest)
+                {
+                    throw new ApplicationException("Stopping further execution as {0} has requested a reboot.".format_with(packageResult.Name));

```

Comment:
We need a way to stop completely if a package has requested for this. I'm wondering if NuGet Service might be a better place to exit the for loop - this is only for install/upgrade if I recall correctly. You will need to find the uninstall result one as well.